src/redux/reducers/servers.test.js

```diff
@@ -0,0 +1,79 @@
+import { Reducer } from 'redux-testkit'
+
+import reducer, { defaultState } from './servers'
+import * as types from './../constants/ActionTypes.js'
+
+describe('viewports reducer', () => {
+  it('should return the initial state', () => {
+    expect(reducer(undefined, {})).toEqual(defaultState)
+  })
+
+  it('should result in one server when there were no servers before on ADD_SERVER', () => {
+    const initialState = defaultState
+
+    const action = { type: types.ADD_SERVER, server: { id: 'some-server-id' } }
+
+    const expectedState = { servers: [action.server] }
+
+    Reducer(reducer)
+      .withState(initialState)
+      .expect(action)
+      .toReturnState(expectedState)
+  })
+
+  it('should add to servers list on ADD_SERVER', () => {
+    const initialState = {
+      servers: [{ id: 'one' }, { id: 'two' }, { id: 'three' }, { id: 'four' }],
+    }
+
+    const action = {
+      type: types.ADD_SERVER,
+      server: { id: 'five' },
+    }
+
+    const expectedState = {
+      servers: [...initialState.servers, action.server],
+    }
+
+    Reducer(reducer)
+      .withState(initialState)
+      .expect(action)
+      .toReturnState(expectedState)
+  })
+
+  it('should not add duplicated servers on ADD_SERVER', () => {
+    const initialState = {
+      servers: [{ id: 'one' }, { id: 'two' }, { id: 'three' }, { id: 'four' }],
+    }
+
+    const action = {
+      type: types.ADD_SERVER,
+      server: { id: 'two' },
+    }
```

Comment:
So the issue here _could be_ that the first instance, if it had additional config properties, could be overridden by the incoming. Even if lodash's `uniqBy` is deterministic, I don't know that it makes and claims of being so.